### DISCUSSION TITLE
AES Key update using microcontroller

### DISCUSSION THREAD

**Discussion by pyanguassierra** [Infineon/linux-optiga-trust-m#59] [General]
Hello. I am trying to provision an AES128 key in the Optiga Trust M directly using a microcontroller, instead of the Linux library and the CLI shown here. I have both ESP32s and a PSoC 6 to test it with, and I would like to not depend on updating the key using a Raspberry Pi as I've been doing. Any help and guidance on whether it is possible to directly provision with a microcontroller and how to do it is greatly appreciated!

**Comment by ying-css** [comment]
Hi @pyanguassierra For AES key protected update use case, there are three steps: 1st step is to set up the metadata for AES key protected update for Trust M(this is called provisioning, it is only needed to be done once), 2nd step is to send the AES key you want to update into Trust M, manifest and fragment to microcontroller, 3rd step is to use the received manifest and fragment to update AES key into Trust M. Please let us know more for your use case in order that we can provide suitable/optimized solution. Thanks a lot.
